service/src/integrationTest/kotlin/fi/espoo/evaka/occupancy/RealtimeOccupancyTest.kt

```diff
@@ -193,6 +193,45 @@ class RealtimeOccupancyTest : FullApplicationTest(resetDbBeforeEach = true) {
             )
         }
     }
+
+    @Test
+    fun testGraphDataShowsNoGapsForOvernightAttendances() {
+        val tomorrow = date.plusDays(1)
+        db.transaction { tx ->
+            FixtureBuilder(tx, date)
+                .addChild()
+                .withAge(2, 10)
+                .saveAnd {
+                    addPlacement().toUnit(testDaycare.id).save()
+                    addAttendance().inUnit(testDaycare.id).arriving(LocalTime.of(20, 45))
+                        .departing(LocalTime.of(23, 59)).save()
+                    addAttendance().inUnit(testDaycare.id).arriving(tomorrow, LocalTime.of(0, 0))
+                        .departing(tomorrow, LocalTime.of(8, 15)).save()
+                }
+                .addEmployee()
+                .withScopedRole(UserRole.STAFF, testDaycare.id)
+                .withGroupAccess(testDaycare.id, groupId)
+                .saveAnd {
+                    addRealtimeAttendance().inGroup(groupId).arriving(LocalTime.of(19, 45))
+                        .departing(tomorrow, LocalTime.of(9, 0)).withCoefficient(occupancyCoefficientSeven)
+                        .withType(StaffAttendanceType.PRESENT).save()
+                }
+        }
+
+        val result = getRealtimeOccupancy()
+        val occupancies = result.occupancySeries
+
+        // 23:59 - 00:00, overnight attendances go to zero in the db during this minute
+        val minuteBeforeMidnight = HelsinkiDateTime.Companion.of(date, LocalTime.of(23, 59))
+        val midnight = HelsinkiDateTime.Companion.of(tomorrow, LocalTime.of(0, 0))
+        val beforeMinuteBeforeMidnight = occupancies.findLast { it.time < minuteBeforeMidnight }
+        occupancies.find { it.time == minuteBeforeMidnight }
+            ?.also { assertEquals(beforeMinuteBeforeMidnight?.childCapacity, it.childCapacity) }
```

Comment:
Using `?.also` in tests is not a good idea because it can make tests pass even when they shouldn't. This test actually fails if you assert that `occupancies.find { it.time == midnight }` is not null.